### ISSUE TITLE
users should be able to set environment variables in cinnamon-session

### ISSUE THREAD

**Issue by brianjmurrell** [linuxmint/cinnamon-session#15]
There appears to be no way for an individual user to be able to set an environment variable in their cinnamon-session (i.e. such that all child processes inherit that variable).

Variables in /etc/environment are set into the users' cinnamon-session but users ought to be able to (a) do this without asking root to help and (b) set variables for only their own session and not every users' sessions.

**Comment by lestcape** [comment]
I agree, i need insert this code to allow the global menu implementation, nemo started so faster and i can not register the environment variables after the start of nemo... so, nemo as a gtk2 app is not called with the correct environment to be translated the appmenu, so, i fail to get a Dbus connection...

```
  g_setenv ("UBUNTU_MENUPROXY", "1", FALSE);
  const gchar *gtk_modules;
  
  gtk_modules = g_getenv ("GTK_MODULES");
  if (gtk_modules != NULL) {
    if(!g_str_match_string("unity-gtk-module", gtk_modules, FALSE)) {
      const gchar * new_gtk_modules = g_strconcat( "unity-gtk-module:", gtk_modules, NULL );
      g_setenv ("GTK_MODULES", new_gtk_modules, TRUE);
    }
  } else {
    g_setenv ("GTK_MODULES", "unity-gtk-module", FALSE);
  }
```

**Comment by Nowaker** [comment]
It doesn't look like adding an env var to `~/.xsessionrc` has an effect on my Cinnamon when using Slim DM. In my case, I had to put the env vars in `~/.xinitrc`.

**Comment by ItzSwirlz** [comment]
Well, what environment variables are you thinking of?

**Comment by PowaBanga** [comment]
GTK_THEME could be a nice one...
